include/maths/analytics/ImportExport.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the following additional limitation. Functionality enabled by the
+ * files subject to the Elastic License 2.0 may only be used in production when
+ * invoked by an Elasticsearch process with a license key installed that permits
+ * use of machine learning features. You may not use this file except in
+ * compliance with the Elastic License 2.0 and the foregoing additional
+ * limitation.
+ */
+#ifndef INCLUDED_ml_maths_analytics_ImportExport_h
+#define INCLUDED_ml_maths_analytics_ImportExport_h
+
+//! On Windows, it's necessary to explicitly export functions from
+//! DLLs.  The simplest way to do this for C++ classes is to add
+//! __declspec(dllexport) between the class keyword and the name
+//! of the class when declaring it.  However, when using the class
+//! from a different DLL or a program, it's necessary that the
+//! __declspec(dllimport) modifier.  Since we only want one header
+//! file declaring each class, this means that these modifiers
+//! have to be held in a macro that expands to __declspec(dllexport)
+//! when the DLL containing the class is being built, but to
+//! __declspec(dllimport) when other code that uses the class is
+//! being built.  On Unix the macro must evaluate to an empty string.
+
+#ifdef Windows
+
+#ifdef BUILDING_libMlMathsAnalytics
+// We're building the maths analytics library
+#define MATHS_ANALYTICS_EXPORT __declspec(dllexport)
+#else
+// We're building some code other than the maths analytics library
+#define MATHS_ANALYTICS_EXPORT __declspec(dllimport)
+#endif
+
+//! See https://devblogs.microsoft.com/cppblog/optimizing-the-layout-of-empty-base-classes-in-vs2015-update-2-3/
+#define EMPTY_BASE_OPT __declspec(empty_bases)
+
+#else
+
+// Empty string on Unix
+#define MATHS_ANALYTICS_EXPORT
+#define EMPTY_BASE_OPT
```

Comment:
Now this is in 3 files, does it cause warnings about defining a macro without undefining it first?  This could be causing build noise on some platforms even if it doesn't happen with clang.

It's also not best practice to be defining the same thing multiple times, so maybe now this is used in multiple libraries it should be moved to `core/WindowsSafe.h`.  Then the classes that need `EMPTY_BASE_OPT` will have to include `core/WindowsSafe.h`, but at least they'll all get the same definition if the way this is done in the future changes.  `STDCALL` is already defined in `core/WindowsSafe.h`, so there's already a precedent.